remote/remote_state.go

```diff
@@ -122,6 +122,11 @@ func (remoteState *RemoteState) differsFrom(existingBackend *TerraformBackend, t
 func (remoteState RemoteState) ToTerraformInitArgs() []string {
 	backendConfigArgs := []string{}
 	for key, value := range remoteState.Config {
+
+		if key == "s3_bucket_tags" || key == "dynamotable_tags" {
+			continue
+		}
```

Comment:
Instead of doing this, couldn't you use `remoteState.Config.remoteStateConfigS3`?

Reply:
Nope, because we are in remote_state.go, not in remote_state_s3.go, where we have this object.

The cleanest would probably be to have the individual backends return the configuration.

Reply:
And I just implemented that, because it does not make sense that the generic remote_state needs to know which keys to filter for remote_state_s3. I encapsulated this there now, and extended the `RemoteStateInitializer` interface by another method `GetTerraformInitArgs`.